markdown_it/rules_core/linkify.py

```diff
@@ -0,0 +1,130 @@
+import re
+
+from ..common.utils import arrayReplaceAt
+from ..common.normalize_url import normalizeLinkText, normalizeLink, validateLink
+from .state_core import StateCore
+from ..token import Token
+
+
+LINK_OPEN_RE = re.compile(r"^<a[>\s]", flags=re.IGNORECASE)
+LINK_CLOSE_RE = re.compile(r"^</a\s*>", flags=re.IGNORECASE)
+
+HTTP_RE = re.compile(r"^http://")
+MAILTO_RE = re.compile(r"^mailto:")
+TEST_MAILTO_RE = re.compile(r"^mailto:", flags=re.IGNORECASE)
+
+
+def isLinkOpen(string: str):
+    return LINK_OPEN_RE.search(string)
+
+
+def isLinkClose(string: str):
+    return LINK_CLOSE_RE.search(string)
+
+
+def linkify(state: StateCore):
+    blockTokens = state.tokens
+
+    if not state.md.options.linkify:
+        return
+
+    if not state.md.linkify:
+        raise Exception("Linkify enabed but not installed.")
```

Comment:
```suggestion
        raise Exception("Linkify enabled but not installed.")
```
There's a typo here which originates from my suggestion. Sorry! Btw, the text was just something I just quickly made up, dont know, maybe we want it to be more elaborate somehow @chrisjsewell ?

Reply:
Also, maybe it could be justified to make the exception type `ModuleNotFoundError` ?

Reply:
or `ImportError`?

I'm worried about exception type. 🤔

Reply:
I think I'd vote for `ModuleNotFoundError` (a subclass of `ImportError`, a bit more precise) as it's equivalent to the root cause of the issue here (the exception we suppress in`markdown_it/main.py` line 16, and never need to raise unless this line runs).